contrib/android/config/android.h

```diff
@@ -26,6 +26,18 @@
 #ifndef ARCHIVE_PLATFORM_H_ANDROID_INCLUDED
 #define	ARCHIVE_PLATFORM_H_ANDROID_INCLUDED
 
+#include <android/api-level.h>
+#ifdef __ANDROID_API__
```

Comment:
If I remember correctly, C standards require that undefined symbols are replaced with 0 when they appear in preprocessor conditionals.

So if __ANDROID_API__ is undefined, then the next line will be processed as #if 0 > 20.

It has been a while since I read that part of the C standard; I hope I'm remembering it correctly.